imap/mailbox.c

```diff
@@ -136,6 +136,7 @@ struct mailbox_iter {
     uint32_t num_records;
     unsigned skipflags;
     seqset_t *uidset;
+    int step_inc;
```

Comment:
nit: this is added to recno later, which is a uint32_t, so maybe step_inc should be int32_t or int8_t.

Reply:
But step_inc can be -1 when iterating backward

Reply:
Yes, that's why I suggest a signed integer value that has less or equal width as an unsigned 32-bit.